Packages/com.unity.inputsystem/InputSystem/Controls/InputControlLayout.cs

```diff
@@ -1243,7 +1243,7 @@ private static NameAndParameters ParseNameAndParameters(string text, ref int ind
             return new NameAndParameters {name = name, parameters = new ReadOnlyArray<ParameterValue>(parameters)};
         }
 
-        private static ParameterValue[] ParseParameters(string parameterString)
+        internal static ParameterValue[] ParseParameters(string parameterString)
```

Comment:
In general, as much as possible, avoid isolating arbitrary pieces from a system for the sake of testing. The tests in `CoreTests_Layouts` should be focused on using layouts as a complete feature. All the CoreTests are integration tests and should not do unit testing.

In this case, I think it's okay-ish. The parameter parsing really should be lifted out of `InputControlLayout` entirely as it has long stopped being specific to just layouts. When that is done, having a unit test suite for just the parameter parsing probably makes sense.

Reply:
I agree that it might make sense to move this logic out, but for now, I changed the test code to use the non-private ParseNameAndParameters method.